_posts/2024-10-26-a-faster-liquidsoap-part-deux.md

```diff
@@ -9,281 +9,310 @@ _The memory part_
+The `2.3.x` release was a great opportunity to catch up with this. It turned out that, with the new script caching, we
+can get initial memory consumption close to the `1.3.x` levels!
 
-But, first, let's look at **how to check memory usage?** Because, well, it turns out that this isn't easy at all..
+But first, let's look at **how to check memory usage**. Because, well, it turns out that this isn't easy at all.
 
 ## The labyrinth of memory allocation
 
-We've had a lot of discussions about how memory is allocated in liquidsoap over the years. [One of the most recent one](https://github.com/savonet/liquidsoap/issues/4058) was
-a great opportunity to realize that it is, in fact, very tricky to actually _measure_ how much memory a process is consuming.
+We've had a lot of discussions about how memory is allocated in Liquidsoap over the years.
+[One of the most recent ones](https://github.com/savonet/liquidsoap/issues/4058) was a great opportunity to realize that
+it is, in fact, very tricky to actually _measure_ how much memory a process is consuming.
 
 Here's a rough rundown of how memory is allocated that might explain that a bit.
 
 ### Memory pages
 
-Internally, memory is organized by _pages_. A page is a fixed-size chunk of memory (usually `4k` on linux, `16k` on Apple Silicon).
+Memory is organized into fixed-size chunks called _pages_. On Linux, a page is typically 4 KB, while on Apple Silicon,
+it is usually 16 KB.
 
-Each process is associated with a collection of memory pages that are assigned to it by the OS kernel. When the process requests some memory to
-be allocated, the OS looks for available memory on the currently allocated pages and, if there is not enough, allocates more pages for the process.
+Each process is associated with a collection of memory pages that are assigned to it by the OS kernel. When the process
+requests memory allocation, the OS looks for available memory on the currently allocated pages and, if there is not
+enough, allocates more pages for the process.
 
-Once a page has some of its memory used by the process, it is considered _dirty_. Before a page can be reclaimed, the process has to free all the memory
-on the page.
+Once a page has had some of its memory used by the process, it is considered _dirty_. Before a page can be reclaimed,
+the process must free all the memory on that page.
 
 ![Paged Memory Illustration](https://github.com/user-attachments/assets/f617f88d-4a7c-4214-bb9d-6d57cfaf3bd5)
 
-Typically, this means that if a process requests `1024` bytes of memory, it is possible that this will result in `4k` of memory being allocated. Likewise,
-if a process releases `1024` bytes of memory, the effective memory it uses might not decrease if the page this chunk of memory was on still has other parts
-of it used.
+When a process requests `1024` bytes of memory, it often results in `4k` bytes being allocated instead. Similarly, if a
```

Comment:
I don't think that this is accurate:
```suggestion
When a process requests `1024` bytes of memory, it may result in `4k` bytes being allocated instead. Similarly, if a
```